src/controls/MenuSelectTableBorderStyle.tsx

```diff
@@ -0,0 +1,193 @@
+  BorderStyleOutset,
+  BorderStyleRidge,
+  BorderStyleSolid,
+} from "../icons";
+import { Z_INDEXES } from "../styles";
+import MenuButton from "./MenuButton";
+import type { MenuButtonTooltipProps } from "./MenuButtonTooltip";
+import MenuControlsContainer from "./MenuControlsContainer";
+
+export type BorderStyleSelectOption = {
+  value: string;
+  IconComponent: React.ElementType<{
+    className: string;
+  }>;
+  label?: string;
+  shortcutKeys?: MenuButtonTooltipProps["shortcutKeys"];
+};
+
+export type MenuSelectTableBorderStyleProps = {
+  options?: BorderStyleSelectOption[];
+  label?: string;
+  className?: string;
+};
+
+const useStyles = makeStyles({ name: { MenuSelectTableBorderStyle } })(
+  (theme) => ({
+    root: {
+      zIndex: Z_INDEXES.BUBBLE_MENU,
+    },
+
+    paper: {
+      backgroundColor: theme.palette.background.default,
+    },
+  })
+);
+
+const DEFAULT_BORDER_STYLE_OPTIONS: BorderStyleSelectOption[] = [
+  {
+    value: "solid",
+    label: "Solid",
+    IconComponent: BorderStyleSolid,
+  },
+  {
+    value: "dashed",
+    label: "Dashed",
+    IconComponent: BorderStyleDashed,
+  },
+  {
+    value: "dotted",
+    label: "Dotted",
+    IconComponent: BorderStyleDotted,
+  },
+  {
+    value: "double",
+    label: "Double",
+    IconComponent: BorderStyleDouble,
+  },
+  {
+    value: "groove",
+    label: "Groove",
+    IconComponent: BorderStyleGroove,
+  },
+  {
+    value: "ridge",
+    label: "Ridge",
+    IconComponent: BorderStyleRidge,
+  },
+  {
+    value: "inset",
+    label: "Inset",
+    IconComponent: BorderStyleInset,
+  },
+  {
+    value: "outset",
+    label: "Outset",
+    IconComponent: BorderStyleOutset,
+  },
+  {
+    value: "none",
+    label: "None",
+    IconComponent: BorderStyleNone,
+  },
+];
+
+export default function MenuSelectTableBorderStyle({
+  options = DEFAULT_BORDER_STYLE_OPTIONS,
+  label,
+  className,
+}: MenuSelectTableBorderStyleProps) {
+  const editor = useRichTextEditorContext();
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
+  const open = Boolean(anchorEl);
+  const { classes, cx } = useStyles();
+  const handleClick = (event: React.MouseEvent<HTMLElement>) => {
+    anchorEl ? handleClose() : setAnchorEl(event.currentTarget);
+  };
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  if (!editor?.isEditable) {
+    return null;
+  }
+
+  const IconComponent =
+    options.find(
+      (option) =>
+        editor.getAttributes("tableHeader").borderStyle === option.value ||
+        editor.getAttributes("tableCell").borderStyle === option.value
+    )?.IconComponent ?? BorderStyle;
+
+  return (
+    <>
+      <MenuButton
+        tooltipLabel={label ?? "Border style"}
+        IconComponent={IconComponent}
+        onClick={handleClick}
+        selected={open}
+        aria-haspopup="true"
+        aria-expanded={open ? "true" : undefined}
+        aria-controls={open ? "border-style-menu" : undefined}
```

Comment:
Thanks for being mindful of accessibility. I don't see the `border-style-menu` id used elsewhere, so I think this attribute isn't correct as is. Presumably we'd need to assign an `id` to the Popper below (assuming I understand `aria-controls` correct) -- but we should probably let the user provide the `id` and not have a fixed value. e.g. if a user includes a `MenuSelectTableBorderStyle` multiple times on the same page, we can't have duplicate IDs. Alternatively we it might be nice to use an auto-generated ID with something like `useId`, but that's only available in React 18, and I'd rather avoid a custom hook or new dep for it.